sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -635,150 +644,86 @@ class ResolveSessionCatalog(
     case _ => throw new AnalysisException(s"$sql is only supported with temp views or v1 tables.")
   }
 
-  private def buildV1Table(
-      ident: TableIdentifier,
-      c: CreateTableAsSelectStatement): Option[CatalogTable] = {
-    val schema = new StructType
-    (c.provider, c.serde) match {
-      case (Some(provider), Some(serde)) =>
-        throw new AnalysisException(
-          s"Cannot create table with both USING $provider and ${serde.describe}")
-
-      case (None, Some(serde)) =>
-        Some(buildHiveCatalogTable(
-          ident, schema, c.partitioning, c.bucketSpec, c.properties, serde, c.options, c.location,
-          c.comment, c.external))
-
-      case (None, None) if !conf.convertCTAS =>
-        Some(buildHiveCatalogTable(
-          ident, schema, c.partitioning, c.bucketSpec, c.properties, SerdeInfo.empty, c.options,
-          c.location, c.comment, c.external))
-
-      case (Some(provider), None) if !isV2Provider(provider) =>
-        Some(buildCatalogTable(
-          ident, schema, c.partitioning, c.bucketSpec, c.properties, provider, c.options,
-          c.location, c.comment, c.external))
-
-      case (None, None) if !isV2Provider(conf.defaultDataSourceName) =>
-        Some(buildCatalogTable(
-          ident, schema, c.partitioning, c.bucketSpec, c.properties, conf.defaultDataSourceName,
-          c.options, c.location, c.comment, c.external))
-
-      case _ =>
-        None
-    }
-  }
-
-  private def buildV1Table(
-      ident: TableIdentifier,
-      c: CreateTableStatement): Option[CatalogTable] = {
-    val schema = c.tableSchema
-    (c.provider, c.serde) match {
-      case (Some(provider), Some(serde)) =>
-        throw new AnalysisException(
-          s"Cannot create table with both USING $provider and ${serde.describe}")
-
-      case (None, Some(serde)) =>
-        Some(buildHiveCatalogTable(
-          ident, schema, c.partitioning, c.bucketSpec, c.properties, serde, c.options, c.location,
-          c.comment, c.external))
-
-      case (None, None) =>
-        Some(buildHiveCatalogTable(
-          ident, schema, c.partitioning, c.bucketSpec, c.properties, SerdeInfo.empty, c.options,
-          c.location, c.comment, c.external))
-
-      case (Some(provider), None) if !isV2Provider(provider) =>
-        Some(buildCatalogTable(
-          ident, schema, c.partitioning, c.bucketSpec, c.properties, provider, c.options,
-          c.location, c.comment, c.external))
-
-      case _ =>
-        None
-    }
-  }
-
-  private def buildCatalogTable(
-      table: TableIdentifier,
-      schema: StructType,
-      partitioning: Seq[Transform],
-      bucketSpec: Option[BucketSpec],
-      properties: Map[String, String],
-      provider: String,
+  private def getStorageFormatAndProvider(
```

Comment:
This method closely follow the original logic of creating hive table in `SparkSqlAstBuilder`